compose_rl/reward_learning/functional.py

```diff
@@ -242,3 +250,190 @@ def __call__(
         curr_rewards = generated_lens / self.max_gen_len
         rewards[torch.arange(batch_size), generated_lens - 1] += curr_rewards
         return rewards
+
+
+class BaseVerifierReward(Reward):
+    # This can be run async
+    BLOCKING = False
+
+    def __init__(self, cfg: dict[Any, Any], tokenizer: Tokenizer):
+        super().__init__(cfg, tokenizer)
+        self.reward = cfg.get('reward', 1.0)
+        log.info(
+            f'Using reward value of {self.reward} for {self.__class__.__name__} verifier',
+        )
+
+    def validate_config(self):
+        # Base validation, can be overridden
+        return
```

Comment:
@bcui-db should validate_config be a member method of `Reward` cls since a bunch of these fn's define it?


Also
`pass` == `return` == `return None`, but `pass` is slightly more explicit for indicating a deliberate no-operation placeholder.
```suggestion
    def validate_config(self):
        # Base validation, can be overridden
        pass
```
maybe I'm just pedantic 🤷‍♂️

Reply:
@vchiley - noted. Will raise a separate PR for this. This Pr is already pretty large, and this will affect several files.